content/articles/save-and-load-stacked-ensembles-in-onnx/index.md

```diff
@@ -0,0 +1,256 @@
+It is also key as a means to avoid vendor lockin since ONNX models can be deployed on any platform - not just where they were trained.
+Container-based methods of pushing models to production like Docker can also be by-passed altogether.
+
+Saving this type of a model in ONNX format and using it in production can prove challenging as this scenario is not adequately documented.
+So how can stacked ensembles be serialized using ONNX?
+
+## Table of Contents
+  1. Preparing the environments
+  2. Importing and preparing the data
+  3. Building and evaluating the classifier
+  4. Serializing the model to ONNX format
+  5. Loading the model using ONNX runtime Inference Session
+
+### Prerequisites
+  1. Basic knowledge of Python
+  2. Machine learning model building, evaluation and validation in Scikit-Learn
+  3. Basic data manipulation skills
+  4. Python(with pip,numpy,pandas and Scikit-Learn) installed on your computer, or an online environment like Google Colab or Kaggle
+
+
+### Goals of the Tutorial
+In this article, you will learn how to:
+  - install ONNX and onnxruntime
+  - determine the ONNX input initialtypes 
+  - serializing and saving a stacked ensemble to ONNX format
+  - loading it to production using an Onnx runtime Inference session.
+
+### Setting up environments
+
+To install ONNX and onnxruntime on a local environment, run the following commands;
+
++ If using pip, on your terminal;
+```
+pip install onnx
+```
+and 
+```
+pip install onnxruntime
+```
+
++ If using anaconda, on anaconda terminal;
+```
+conda install -c conda-forge onnx
+```
+and
+```
+conda install -c conda-forge onnxruntime
+```
+
+Note: ONNX is not pre-installed in the runtime environments on Google Colab and Kaggle notebooks
+
+To install ONNX and onnxruntime on Google Colab or Kaggle, run;
+```
+!pip install onnx
+```
+and
+```
+!pip install onnxruntime
+```
+**Note:** Online editors like repl.it may fail to run the code below in full due to in-sufficient memory allocations.
+
+### Importing and preparing the data
+Let's start by importing pandas and the data
+```python
+import pandas as pd
+path='https://raw.githubusercontent.com/iannjari/datasets/main/diabetes.csv'
+df=pd.read_csv(path,engine='python')
+df
+```
+Output:
+
+![DataFrame](df1.png)
+*Image by author*
+
+We will seperate the data columns from the label/target, "Outcome";
+
+```python
+target_name = "Outcome"
+target = df[target_name]
+
+data = df.drop(columns=[target_name])
+data
+```
+Split the data into training and testing partitions;
+
+```python
+from sklearn.model_selection import train_test_split
+x_train, x_test, y_train, y_test = train_test_split(
+    data,target, test_size=0.33, random_state=42)
+```
+
+### Training and evaluating the stacked classifier
+We shall employ a stack with a randomForest, kNN classifier, GradientBoosting classifier and a logistic regressor as a final model.
+Import packages;
```

Comment:
Add links that speak about these models and give a single line explanation of each